driver-core/src/main/java/com/datastax/driver/core/Frame.java

```diff
@@ -175,7 +178,10 @@ static int lengthFor(ProtocolVersion version) {
                 return set;
             }
 
-            static int serialize(EnumSet<Flag> flags) {
+            static int serialize(EnumSet<Flag> flags, ProtocolVersion protocolVersion) {
+                if (protocolVersion.toInt() < ProtocolVersion.V5.toInt() && flags.contains(USE_BETA))
```

Comment:
I think we have a problem here. `Frame.BetaVersionFlagAdder` is executed _after_  `Frame.Encoder` so the flag will never be set at this point. Maybe a better approach is to move this check to `Frame.BetaVersionFlagAdder`.